distribution/lib/Standard/Base/0.0.0-dev/src/Data/Time/Date.enso

```diff
@@ -336,3 +336,9 @@ type Date
     compare_to that =
         sign = self.internal_local_date.compareTo that.internal_local_date
         Ordering.from_sign sign
+
+    ## Compares two Dates for equality.
+    == : Date -> Boolean
+    == that = case that of
+        Date _ -> self.internal_local_date.equals that.internal_local_date
+        _ -> False
```

Comment:
I wonder if after https://github.com/enso-org/enso/pull/3568 this will still be needed.

Regardless of that, I think it would be good to somewhere add tests for:
```
a = LocalDate.of 1990 10 10
b = LocalDate.of 1990 10 10
c = LocalDate.of 1990 10 10

a==a . should_be_true
a==b . should_be_true
a==c . should_be_false
```

Currently (checked on `develop`, not on this branch), even `a==a` would evaluate to False because somehow magically the first `a` is an Atom but the second one is a Polyglot value and the implementation of `==` gets confused. So it's worth checking if after these changes this works properly.

Reply:
@radeusgd, can you please share a real program with me? The code snippet you have provided doesn't even compile. If I modify it to:
```
polyglot java import java.time.LocalDate

main =
    a = LocalDate.of 1990 10 10
    b = LocalDate.of 1990 10 10
    c = LocalDate.of 1990 10 10

    a==a . should_be_true
    a==b . should_be_true
    a==c . should_be_false
```
ensure scope exists - [fc2dca2](https://github.com/enso-org/enso/pull/3559/commits/fc2dca2cfdb415d7edd713526bb427c78a9a9eed) - and execute it I get:
```
Execution finished with an error: Method `==` of java.time.LocalDate could not be found.
        at <enso> argument<0>(d.enso:9:5-8)
        at <enso> d.main(d.enso:9:5-25)
```

Reply:
Following program runs OK now:
```
from Standard.Base import all
import Standard.Test
polyglot java import java.time.LocalDate

main =
    a = LocalDate.of 1990 10 10
    b = LocalDate.of 1990 10 10
    c = LocalDate.of 1990 10 10

    a==a . should_be_true
    a==b . should_be_true
    a==c . should_be_true
```
however the program is different to your one, Radek. I am not sure what was your intention with `a==c . should_be_false`?